lib/chef/knife/winrm_knife_base.rb

```diff
@@ -122,11 +121,15 @@ def relay_winrm_command(command)
 
             queue = Queue.new
             @winrm_sessions.each { |s| queue << s }
+            con_limit = config[:concurrency]
+            num_sessions =  locate_config_value(:concurrency)
+            num_targets =  @winrm_sessions.length
+            num_sessions  = (con_limit.nil? || con_limit == 0) ? num_targets : [num_sessions, num_targets].min
```

Comment:
A little confused by this - doesn't `locate_config_value` also search in `config`?

Reply:
Actually it is but setting con_limit as hard code value might be not a good idea, So set it as `config[:concurrency]`

Reply:
What I mean is, is there a scenario in which `locate_config_value` will fail to return the value included in `config[:concurrency]`?

Reply:
Not require to set con_limit as `config[:concurrency]` will be taken care in def locate_config_value,
made changes as required.